shell/platform/android/io/flutter/embedding/engine/FlutterJNI.java

```diff
@@ -272,36 +317,75 @@ public long performNativeAttach(@NonNull FlutterJNI flutterJNI, boolean isBackgr
 
   private native long nativeAttach(@NonNull FlutterJNI flutterJNI, boolean isBackgroundView);
 
+  /**
+   * Spawns a new FlutterJNI instance from the current instance.
+   *
+   * <p>This creates another native shell from the current shell. This causes the 2 shells to re-use
+   * some of the shared resources, reducing the total memory consumption versus creating a new
+   * FlutterJNI by calling its standard constructor.
+   *
+   * <p>This can only be called once the current FlutterJNI instance is attached by calling {@link
+   * #attachToNative(boolean)}.
+   *
+   * <p>Static methods that should be only called once such as {@link #init(Context, String[],
+   * String, String, String, long)} or {@link #setRefreshRateFPS(float)} shouldn't be called again
+   * on the spawned FlutterJNI instance.
+   */
+  @UiThread
+  @NonNull
+  public FlutterJNI spawn(
+      @Nullable String entrypointFunctionName, @Nullable String pathToEntrypointFunction) {
+    ensureRunningOnMainThread();
+    ensureAttachedToNative();
+    FlutterJNI spawnedJNI = new FlutterJNI();
+    // Call the native function with the current Shell ID. It creates a new Shell. Feed the new
+    // Shell ID into the new FlutterJNI instance.
+    spawnedJNI.nativeShellId =
+        nativeSpawn(spawnedJNI, nativeShellId, entrypointFunctionName, pathToEntrypointFunction);
+    Preconditions.checkState(
```

Comment:
This name is kind of silly, in the case it's a post-condition =)

Reply:
Ya, the name is a bit silly. That's just what Java/Guava calls these things. cs/Preconditions.checkState